File: LENGUAJES/PHP/Ejercicio13.php
<?php
// OCTAVOS DE FINAL

function simularPartido($equipo1, $equipo2) {
    /** Simula un partido entre dos equipos y devuelve el ganador. */
    return rand(0, 1) ? $equipo1 : $equipo2;
}

function octavosDeFinal($equipos) {
    /** Simula los octavos de final con los equipos dados. */
    echo "Octavos de Final:\n";
    $ganadores = [];

    // Asegurarse de que hay un número par de equipos
    if (count($equipos) % 2 != 0) {
        echo "El número de equipos debe ser par.\n";
        return;
    }

    for ($i = 0; $i < count($equipos); $i += 2) {
        $equipo1 = $equipos[$i];
        $equipo2 = $equipos[$i + 1];
        $ganador = simularPartido($equipo1, $equipo2);
        $ganadores[] = $ganador;
        echo "Partido: $equipo1 vs $equipo2 - Ganador: $ganador\n";
    }

    return $ganadores;
}

function main() {
    // Definimos los equipos que participan en los octavos de final
    $equipos = [
        "FC Barcelona",
        "Real Madrid",
        "Arsenal",
        "Manchester City",
        "Chelsea",
        "Liverpool",
        "PSG",
        "Benfica"
    ];

    // Simular los octavos de final
    $ganadores = octavosDeFinal($equipos);

    echo "\nEquipos que avanzan a los cuartos de final:\n";
    foreach ($ganadores as $ganador) {
        echo "$ganador\n";
    }
}
main();
?>

File: LENGUAJES/GO/Ejercicio13.go
//OCTAVOS DE FINAL
package main

import (
	"fmt"
	"math/rand"
	"time"
)

func simularPartido(equipo1, equipo2 string) string {
	if rand.Intn(2) == 0 {
		return equipo1
	}
	return equipo2
}

func octavosDeFinal(equipos []string) []string {
	fmt.Println("Octavos de Final:")
	ganadores := []string{}

	if len(equipos)%2 != 0 {
		fmt.Println("El número de equipos debe ser par.")
		return nil
	}

	for i := 0; i < len(equipos); i += 2 {
		equipo1 := equipos[i]
		equipo2 := equipos[i+1]
		ganador := simularPartido(equipo1, equipo2)
		ganadores = append(ganadores, ganador)
		fmt.Printf("Partido: %s vs %s - Ganador: %s\n", equipo1, equipo2, ganador)
	}

	return ganadores
}

func main() {
	rand.Seed(time.Now().UnixNano())

	equipos := []string{
		"FC Barcelona",
		"Real Madrid",
		"Arsenal",
		"Manchester City",
		"Chelsea",
		"Liverpool",
		"PSG",
		"Benfica",
	}

	ganadores := octavosDeFinal(equipos)

	fmt.Println("\nEquipos que avanzan a los cuartos de final:")
	for _, ganador := range ganadores {
		fmt.Println(ganador)
	}
}